core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java

```diff
@@ -63,8 +63,8 @@ public static Envelope from(
             for (var speedRange : subMap.asMapOfRanges().entrySet()) {
                 // compute where this limit is active from and to
                 var interval = speedRange.getKey();
-                var begin = offset + interval.lowerEndpoint();
-                var end = offset + interval.upperEndpoint();
+                var begin = offset + interval.lowerEndpoint() + driverBehaviour.getSpeedTransitionOffsetIncreasing();
+                var end = offset + interval.upperEndpoint() + driverBehaviour.getSpeedTransitionOffsetDecreasing();
```

Comment:
The direction of the offset and the offset itself must depend on the speed transition.
If the speed transition is increasing, offset the transition with positive values, because the driver postpones the acceleration.
If the speed transition is decresing, offset with negative value, because the driver anticipates the braking.

Reply:
resolved